source/code/omi_preexec/mycnf-scanner.cpp

```diff
@@ -0,0 +1,251 @@
+                    {
+                        SCX_LOGHYSTERICAL(m_log, L"   New Section, scanning ...");
+
+                        // Starting new section - set defaults
+                        portNum = 3306;
+                        bindAddress = L"127.0.0.1";
+                    }
+                    continue;
+                }
+
+                std::vector<std::wstring> elements;
+                SCXCoreLib::StrTokenize(line, elements, L"=");
+                if ( elements.size() != 2 )
+                {
+                    // Not a line that we care about
+                    continue;
+                }
+
+                // Check for a port declaration (port = xxx)
+                if ( 0 == SCXCoreLib::StrCompare(L"port", elements[0], true) )
+                {
+                    SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"   Found Port: ", elements[1]));
+                    portNum = SCXCoreLib::StrToUInt( elements[1] );
+                }
+
+                // Check for a bind-address declaration (bind-address = 172.0.0.1)
+                if ( 0 == SCXCoreLib::StrCompare(L"bind-address", elements[0], true) )
+                {
+                    SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"   Found bind-address: ", elements[1]));
+                    bindAddress = elements[1];
+                }
+            }
+
+            if ( portNum )
+            {
+                SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(SCXCoreLib::StrAppend(L"   Flushing port ", portNum).append(L", Bind Address: "), bindAddress));
+
+                auth.AddCredentialSet(portNum, SCXCoreLib::StrToUTF8(bindAddress));
+                m_ports[portNum] = bindAddress;
+            }
+        }
+    }
+
+    /**
+       call LoadCnfFile on each ".cnf" file present in the given directory.
+
+       \param[in]  directory                     path of the MySQL configuration file
+       \param[in]  auth                          authentication object that has been 'Load'ed.
+       \throws     SCXFilePathNotFoundException  if directory is not found
+    */
+    void MySQL_ConfigurationScanner:: LoadCnfDirectory(
+        const SCXCoreLib::SCXFilePath& directory, 
+        MySQL_Authentication& auth)
+    {
+        SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"LoadCnfFilesFromDirectory", directory.Get()));
+        std::vector< SCXCoreLib::SCXFilePath > cnfFiles = m_extDepends->GetFiles(directory);
+
+        for( std::vector< SCXCoreLib::SCXFilePath >::const_iterator iter = cnfFiles.begin(); iter != cnfFiles.end(); ++iter)
+        {
+            std::wstring fileSuffix = iter->GetFilesuffix();
+            if (fileSuffix == L"cnf")
+            {
+                LoadCnfFile(*iter, auth);
+            }
+            else
+            {
+                SCX_LOGHYSTERICAL(m_log,SCXCoreLib::StrAppend(iter->Get(), L" is not a cnf file"));
+            }
```

Comment:
This seems needlessly restrictive. Does MySQL itself allow config files to be named anything (like MySQL.config?)  Is this a MySQL restriction, or are you imposing this yourself?

Also, what happens if an include directory contains other directories (even like "secondinclude.cnf")? It would pass your test. You probably shouldn't call LoadCnfFile on a directory, or anything that isn't a regular file for that matter.